app/controllers/api/storage.php

```diff
@@ -306,97 +307,200 @@
+                    $data = OpenSSL::encrypt($data, OpenSSL::CIPHER_AES_128_GCM, $key, 0, $iv, $tag);
+                }
+                if (!$device->write($path, $data, $mimeType)) {
+                    throw new Exception('Failed to save file', 500);
+                }
+            }
+
+
+            $sizeActual = $device->getFileSize($path);
+
+            $read = (is_null($read) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $read ?? [];
+            $write = (is_null($write) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $write ?? [];
+            $algorithm = empty($compressor) ? '' : $compressor->getName();
+            $fileHash = $device->getFileHash($path);
+
+            if ($bucket->getAttribute('encryption', true) && $size <= APP_LIMIT_ENCRYPTION) {
+                $openSSLVersion = '1';
+                $openSSLCipher = OpenSSL::CIPHER_AES_128_GCM;
+                $openSSLTag = \bin2hex($tag);
+                $openSSLIV = \bin2hex($iv);
+            }
+
+            if ($file->isEmpty()) {
+                $data = [
+                    '$read' => $read,
+                    '$write' => $write,
+                    'dateCreated' => \time(),
+                    'bucketId' => $bucket->getId(),
+                    'name' => $fileName,
+                    'path' => $path,
+                    'signature' => $fileHash,
+                    'mimeType' => $mimeType,
+                    'sizeOriginal' => $size,
+                    'sizeActual' => $sizeActual,
+                    'algorithm' => $algorithm,
+                    'comment' => '',
+                    'totalChunks' => $chunks,
+                    'uploadedChunks' => $uploadedChunks,
+                    'openSSLVersion' => $openSSLVersion,
+                    'openSSLCipher' => $openSSLCipher,
+                    'openSSLTag' => $openSSLTag,
+                    'openSSLIV' => $openSSLIV,
+                ];
+                $file = $dbForInternal->createDocument('files', new Document($data));
+            } else {
+                $file = $dbForInternal->updateDocument('files', $uploadId, $file
+                        ->setAttribute('$read', $read)
+                        ->setAttribute('$write', $write)
+                        ->setAttribute('signature', $fileHash)
+                        ->setAttribute('mimeType', $mimeType)
+                        ->setAttribute('sizeActual', $sizeActual)
+                        ->setAttribute('algorithm', $algorithm)
+                        ->setAttribute('openSSLVersion', $openSSLVersion)
+                        ->setAttribute('openSSLCipher', $openSSLCipher)
+                        ->setAttribute('openSSLTag', $openSSLTag)
+                        ->setAttribute('openSSLIV', $openSSLIV)
+                );
+            }
+        } else {
+            if ($file->isEmpty()) {
+                $data = [
```

Comment:
You don't really need this var. You can just create the document instance right here.